reactor-netty5-http/src/main/java/reactor/netty5/http/client/HttpClientOperations.java

```diff
@@ -782,6 +791,120 @@ static final class ResponseState {
 		}
 	}
 
+	static final class SendForm extends Mono<Void> {
+
+		static final HttpDataFactory DEFAULT_FACTORY = new DefaultHttpDataFactory(DefaultHttpDataFactory.MINSIZE);
+
+		final HttpClientOperations                                  parent;
+		final BiConsumer<? super HttpClientRequest, HttpClientForm> formCallback;
+		final Consumer<Flux<Long>>                                  progressCallback;
+
+		SendForm(HttpClientOperations parent,
+				BiConsumer<? super HttpClientRequest, HttpClientForm>  formCallback,
+				@Nullable Consumer<Flux<Long>> progressCallback) {
+			this.parent = parent;
+			this.formCallback = formCallback;
+			this.progressCallback = progressCallback;
+		}
+
+		@Override
+		public void subscribe(CoreSubscriber<? super Void> s) {
+			if (!parent.markSentHeaders()) {
+				Operators.error(s,
+						new IllegalStateException("headers have already been sent"));
+				return;
+			}
+			Subscription subscription = Operators.emptySubscription();
+			s.onSubscribe(subscription);
+			if (parent.channel()
+			          .executor()
+			          .inEventLoop()) {
+				_subscribe(s);
+			}
+			else {
+				parent.channel()
+				      .executor()
+				      .execute(() -> _subscribe(s));
+			}
+		}
+
+		void _subscribe(CoreSubscriber<? super Void> s) {
+			HttpDataFactory df = DEFAULT_FACTORY;
+
+			try {
+				HttpClientFormEncoder encoder = new HttpClientFormEncoder(df,
+						parent.nettyRequest,
+						false,
+						HttpConstants.DEFAULT_CHARSET,
+						HttpPostRequestEncoder.EncoderMode.RFC1738);
+
+				formCallback.accept(parent, encoder);
+
+				encoder = encoder.applyChanges(parent.nettyRequest);
+				df = encoder.newFactory;
+
+				if (!encoder.isMultipart()) {
+					parent.requestHeaders.remove(HttpHeaderNames.TRANSFER_ENCODING);
+				}
+
+				parent.addHandlerFirst(NettyPipeline.ChunkedWriter, new ChunkedWriteHandler());
+
+				boolean chunked = HttpUtil.isTransferEncodingChunked(parent.nettyRequest);
+
+				HttpRequest r = encoder.finalizeRequest();
+
+				if (!chunked) {
+					HttpUtil.setTransferEncodingChunked(r, false);
+					HttpUtil.setContentLength(r, encoder.length());
+				}
+
+				Mono<Void> mono = Mono.fromCompletionStage(parent.channel()
```

Comment:
Let's create the `Mono` if it is needed. This happens only when `chunked` is disabled.

```suggestion
				Future<Void> f = parent.channel().writeAndFlush(r);
```

Reply:
applied feedback in commit [cb44675](https://github.com/reactor/reactor-netty/pull/2327/commits/cb44675e86eff8e5a5427adafcd2e75e6892126e)